scripting/executes.sp

```diff
@@ -1106,10 +1118,12 @@ public void UpdateTeams() {
 
       if (StrEqual(g_LastItemPickup[client], "ak47")) {
         g_PlayerPrimary[client] = "weapon_ak47";
-      } else if (g_SilencedM4[client]) {
+      } else if (g_CTRifle[client] == CTRiflePref_M4) {
+        g_PlayerPrimary[client] = "weapon_m4a1";
```

Comment:
I think this type of logic (same with the T side) should be encapsulated into another function `GivePlayerPrimaryRifle` or such. That way this logic doesn't need to be repeated so much.